app/models/miq_report/formatters/graph.rb

```diff
@@ -11,6 +11,18 @@ def graph_options(options = nil)
 
   def to_chart(theme = nil, show_title = false, graph_options = nil)
     ManageIQ::Reporting::Formatter::ReportRenderer.render(ManageIQ::Reporting::Charting.format) do |e|
+      for i in 0 ... col_formats.size do
+        if [:bytes_human_precision_2, :kilobytes_human, :megabytes_human, :megabytes_human_precision_2, :gigabytes_human, :gigabytes_human_precision_2].include? col_formats[i]
+          final_unit = 1
+          table.data.each do |d|
+            unit = [1.terabytes, 1.gigabytes, 1.megabytes, 1.kilobytes].detect { |u| d[i] / u > 10 }
+            final_unit = unit if unit > final_unit
+          end
+          table.data.each do |d|
+            d[i] /= final_unit
+          end
+        end
+      end
```

Comment:
Prefer iterating the col_formats themselves as opposed to iterating the indexes:

```suggestion
      col_formats.each_with_index do |format, i|
        if [:bytes_human_precision_2, :kilobytes_human, :megabytes_human, :megabytes_human_precision_2, :gigabytes_human, :gigabytes_human_precision_2].include?(format)
          final_unit = 1
          table.data.each do |d|
            unit = [1.terabytes, 1.gigabytes, 1.megabytes, 1.kilobytes].detect { |u| d[i] / u > 10 }
            final_unit = unit if unit > final_unit
          end
          table.data.each do |d|
            d[i] /= final_unit
          end
        end
      end
```

Reply:
There is a general helper to do the conversion for you in NumberHelper, but I'd have to see how to apply it.